lib/download.go

```diff
@@ -136,18 +136,37 @@ func downloadFromURL(installLocation string, url string, wg *sync.WaitGroup) (st
 
 func downloadPublicKey(product Product, installLocation string, wg *sync.WaitGroup) (string, error) {
 	pubKeyFilePath := filepath.Join(installLocation, "/", product.GetId()+"_"+product.GetPublicKeyId()+pubKeySuffix)
-	logger.Debugf("Looking up public key file at %q", pubKeyFilePath)
+	logger.Debugf("Looking up public PGP-key file at %q", pubKeyFilePath)
 	publicKeyFileExists := FileExistsAndIsNotDir(pubKeyFilePath)
 	if !publicKeyFileExists {
-		// Public key does not exist. Let's grab it from hashicorp
-		pubKeyFile, errDl := downloadFromURL(installLocation, product.GetPublicKeyUrl(), wg)
-		if errDl != nil {
-			logger.Errorf("Error fetching public key file from %s", product.GetPublicKeyUrl())
-			return "", errDl
+		// Public PGP-key does not exist. Let's grab it
+		publicKeyURLs := product.GetPublicKeyURLs()
+		var pubKeyFile string
+		var errDl error
+		var errsDl []string
```

Comment:
I imagine this shouldl be a list of errors - I think you can do some clever things with them: to combine them: https://pkg.go.dev/errors